swarm/runtime/statsdb/ingestion.py

```
@@ -549,171 +549,230 @@ def ingest_events(self, events: List[Dict[str, Any]], run_id: str) -> int:
+                    processed_ids = set()
+                    for event in events:
+                        event_id = event.get("event_id")
+                        if event_id in new_ids and event_id not in processed_ids:
+                            self._process_event_projections(event, run_id)
+                            processed_ids.add(event_id)
+
+                    # 6. Cleanup
+                    self.connection.execute("DROP TABLE batch_events_staging")
+
+                    self.connection.execute("COMMIT")
+                    return newly_ingested
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)